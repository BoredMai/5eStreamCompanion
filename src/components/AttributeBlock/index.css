:local(.attribute-block) {
  background: no-repeat url('/src/assets/images/stat_scroll.png') 50% / 100%;
  image-rendering: pixelated;
  width: 8.3rem;
  height: 10.3rem;
}

:local(.attribute-block .bonus) {
  font-size: 3.2rem;
  text-align: center;
}

:local(.attribute-block .score) {
  position: relative;
  top: -0.25rem;
}
