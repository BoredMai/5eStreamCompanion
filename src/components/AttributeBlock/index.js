import cn from 'classnames';
import React, { Component } from 'react';
import PropTypes from 'prop-types';

import Attributes from '../../constants/attributes';

import { attributeBlock, bonus, score as attrScore } from './index.css';
import { tag, score, whiteOutline } from '../../index.css';

export default class AttributeBlock extends Component {
  render() {
    const { attribute, value } = this.props;

    return (
      <div className={attributeBlock}>
        <div className={cn(tag, whiteOutline)}>{attribute}</div>
        <div className={cn(attrScore, score, whiteOutline)}>{value}</div>
        <div className={cn(bonus, whiteOutline)}>
          {value > 9 && '+'}
          {Math.floor((value - 10) / 2)}
        </div>
      </div>
    );
  }

  static propTypes = {
    attribute: PropTypes.oneOf(Object.keys(Attributes)).isRequired,
    value: PropTypes.number.isRequired,
  };
}
