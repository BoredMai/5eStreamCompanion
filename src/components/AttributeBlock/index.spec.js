import React from 'react';
import { shallow } from 'enzyme';
import AttributeBlock from '.';

describe('<AttributeBlock />', () => {
  let wrapper;
  const props = {
    attribute: 'STR',
    value: 20,
  };

  beforeEach(() => {
    wrapper = shallow(<AttributeBlock {...props} />);
  });

  it('renders properly', () => {
    expect(wrapper.getElement()).toMatchSnapshot();
  });

  it('renders correct bonus value', () => {
    const bonus = wrapper.find('div.bonus');
    expect(bonus).toHaveLength(1);
    expect(bonus.text()).toBe('+5');
  });

  it('renders correct bonus value on numbers under 10', () => {
    wrapper = shallow(<AttributeBlock {...props} value={9} />);
    const bonus = wrapper.find('div.bonus');
    expect(bonus).toHaveLength(1);
    expect(bonus.text()).toBe('-1');
  });
});
