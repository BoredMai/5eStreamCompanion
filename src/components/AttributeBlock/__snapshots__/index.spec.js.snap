// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AttributeBlock /> renders properly 1`] = `
<div
  className="attributeBlock"
>
  <div
    className="tag whiteOutline"
  >
    STR
  </div>
  <div
    className="score whiteOutline"
  >
    20
  </div>
  <div
    className="bonus whiteOutline"
  >
    +
    5
  </div>
</div>
`;
